src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke.util;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+public class TaskList {
+    private ArrayList<Task> arr;
+
+    /**
+     * Constructor of class TaskList.
+     * @param arr The input ArrayList that stores the past tasks that were stored
+     *            in a file if there were any.
+     */
+    public TaskList(ArrayList<Task> arr) {
+        // arr is given from the duke.util.Storage.load()
+        this.arr = arr;
+    }
+
+    /**
+     * To find tasks containing the String keyword.
+     * @param keyword The input keyword.
+     * @return An ArrayList containing the tasks containing the keyword.
+     */
+    public ArrayList<Task> find(String keyword) {
+        ArrayList<Task> containsKey = new ArrayList<>();
```

Comment:
would it be better to name this ArrayList of tasks, tasksContainingkey for it to be more intuitive?